lib/ansiblelint/formatters/__init__.py

```diff
@@ -42,6 +44,9 @@ def _format_path(self, path: Union[str, Path]) -> str:
         # Use os.path.relpath 'cause Path.relative_to() misbehaves
         return os.path.relpath(path, start=self._base_dir)
 
+    def format(self, match: "MatchError", colored: bool = False) -> str:
+        return str(match)
+
```

Comment:
I don't think this is related to reducing the complexity in main

Reply:
It forces inherited objects to respect type, this change is as much about typing as reducing complexity. I will make it clear in description, as original commit message did not endup being copied and the subject line is too short.